src/components/ListItem.tsx

```diff
@@ -98,7 +98,9 @@ const ListItem = ({
   disabled = false
 }: ListItemProps): ReactElement => {
   const [pressInY, setPressInY] = useState<number | null>(null)
-  const [pressed, setPressed] = useState<boolean>(false)
+  const [pressed, setPressedState] = useState<boolean>(false)
+  const setPressed = (pressed: boolean): void => onPress && setPressedState(pressed)
```

Comment:
Please find another name for current `setPressed` and rename the `useState` method to `setPressed`. I think this is a convention we should stick to that `useState` value and setter are always the same. How about or similar:
```suggestion
  const [pressed, setPressed] = useState<boolean>(false)
  const updatePressed = (pressed: boolean): void => onPress && setPressedState(pressed)
```
I also think it should depend on `disabled` instead of on `onPress`. If the button is not disabled it should react to user presses even if there is no `onPress` method. At least that is what I'd expect.

Reply:
My thoughts about this were:
1) you don't have to think that setPress needs different handling and just call it as normal and side-case handling is done for you
2) disabling changes the color too, and we what three types of behaviour: a) disabled = grey and no action on click b) disabled = not-grey but no action on click c) not disabled = not-grey and action on click.